astropy/utils/masked/function_helpers.py

```diff
@@ -1065,11 +1074,16 @@ def nanfunc(a, *args, **kwargs):
     return nanfunc
 
 
-for nanfuncname in np.lib.nanfunctions.__all__:
-    globals()[nanfuncname] = dispatched_function(
-        masked_nanfunc(nanfuncname), helps=getattr(np, nanfuncname)
-    )
-
+if NUMPY_LT_2_0:
+    for nanfuncname in np.lib.nanfunctions.__all__:
+        globals()[nanfuncname] = dispatched_function(
+            masked_nanfunc(nanfuncname), helps=getattr(np, nanfuncname)
+        )
+else:
+    for nanfuncname in np.lib._nanfunctions_impl.__all__:
```

Comment:
This seems weird - they must still be publicly available! I'll check...

Reply:
Ah, no, I misunderstood: here we just get the names out, the functions themselves are under `np`, of course.